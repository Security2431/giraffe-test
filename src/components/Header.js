import React from "react";
import AuthForm from "./AuthForm";

const styles = {
  headerStyle: {
    padding: "15px 0",
    borderBottom: "1px solid #eee"
  }
};

const Header = ({ isAuthenticated, activeUser, auth, logout }) => {
  return (
    <header style={styles.headerStyle}>
      <div className="container-fluid">
        <div className="row justify-content-end align-items-center">
          {isAuthenticated && (
            <div className="col-auto">
              <button className="btn btn-success">Create Ad</button>
            </div>
          )}

          <div className="col-auto">
            <AuthForm
              isAuthenticated={isAuthenticated}
              activeUser={activeUser}
              onSubmit={auth}
              logout={logout}
            />
          </div>
        </div>
      </div>
    </header>
  );
};

export default Header;
