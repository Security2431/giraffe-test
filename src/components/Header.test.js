/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

jest.mock('./AdListItem', () => () => 'AdListItem');
jest.mock('./AdListPagination', () => () => 'AdListPagination');

const mockProps = {
  auth: jest.fn(),
  logout: jest.fn(),
  activeUser: 'Admin',
  isAuthenticated: true,
};

describe('<Header />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Header {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header {...mockProps} />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
