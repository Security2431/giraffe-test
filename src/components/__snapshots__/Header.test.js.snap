// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<header
  style={
    Object {
      "borderBottom": "1px solid #eee",
      "padding": "15px 0",
    }
  }
>
  <div
    className="container-fluid"
  >
    <div
      className="row justify-content-end align-items-center"
    >
      <div
        className="col-auto"
      >
        <a
          href="/edit"
          onClick={[Function]}
        >
          <button
            className="btn btn-danger"
            type="button"
          >
            Create Ad
          </button>
        </a>
      </div>
      <div
        className="col-auto"
      >
        <div
          className="row align-items-center"
        >
          <div
            className="col"
          >
            <b>
              Admin
            </b>
          </div>
          <div
            className="col"
          >
            <button
              className="btn btn-link"
              onClick={[MockFunction]}
              type="button"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                LogÂ out
              </a>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
